protocol/communities_messenger_signers_test.go

```diff
@@ -154,53 +160,294 @@ func (s *MessengerCommunitiesSignersSuite) TestControlNodeUpdateSigner() {
 	s.Require().NoError(err)
 	s.Require().True(community.IsControlNode())
+	s.Require().True(common.IsPubKeyEqual(community.ControlNode(), &s.alice.identity.PublicKey))
+	s.Require().True(community.IsOwner())
 
-	// john accepts community update from alice (new control node)
+	// check that Bob received kick event, also he will receive
+	// request to share RevealedAddresses and send request to join to the control node
+	_, err = WaitOnSignaledMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && !r.Communities()[0].HasMember(&s.bob.identity.PublicKey)
+		},
+		"Bob was not kicked from the community",
+	)
+	s.Require().NoError(err)
+
+	// check that John received kick event, also he will receive
+	// request to share RevealedAddresses and send request to join to the control node
 	_, err = WaitOnSignaledMessengerResponse(
 		s.john,
 		func(r *MessengerResponse) bool {
-			return len(r.Communities()) > 0 && r.Communities()[0].IDString() == community.IDString()
+			return len(r.Communities()) > 0 && !r.Communities()[0].HasMember(&s.john.identity.PublicKey)
 		},
-		"no communities",
+		"John was not kicked from the community",
 	)
 	s.Require().NoError(err)
 
-	// We check the control node is correctly set on john to alice
-	johnCommunity, err := s.john.communitiesManager.GetByIDString(community.IDString())
+	// Alice auto-accept requests to join with RevealedAddresses
+	// TODO: please, check TODO's in this test and uncomment them if Members() == 3
+	_, err = WaitOnMessengerResponse(
+		s.alice,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && len(r.Communities()[0].Members()) == 2
+		},
+		"no community update with accepted request",
+	)
 	s.Require().NoError(err)
-	s.Require().NotNil(johnCommunity)
-	s.Require().True(common.IsPubKeyEqual(johnCommunity.ControlNode(), &s.alice.identity.PublicKey))
-	s.Require().False(johnCommunity.IsControlNode())
+
+	validateResults := func(messenger *Messenger) *communities.Community {
+		community, err = messenger.communitiesManager.GetByID(community.ID())
+		s.Require().NoError(err)
+		s.Require().True(common.IsPubKeyEqual(community.ControlNode(), &s.alice.identity.PublicKey))
+		s.Require().Len(community.Members(), 2)
+		s.Require().True(community.HasMember(&messenger.identity.PublicKey))
+
+		return community
+	}
+
+	community = validateResults(s.alice)
+	s.Require().True(community.IsControlNode())
+	s.Require().True(community.IsOwner())
+
+	// Bob is a community member again
+	_, err = WaitOnMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && r.Communities()[0].HasMember(&s.bob.identity.PublicKey)
+		},
+		"Bob was auto-accepted",
+	)
+	s.Require().NoError(err)
+
+	community = validateResults(s.bob)
+	s.Require().False(community.IsControlNode())
+	s.Require().False(community.IsOwner())
+
+	// TODO: uncomment when ex-owner will start sharing request to join with revealed address
```

Comment:
Shouldn't it happen already with improved implementation of listening to control node change?

Reply:
No, the community owner does not keep in the database his own request to join with revealed accounts. We discussed it on Friday meeting, I created a ticket to solve it, it on me